src/goto-programs/show_goto_functions.cpp

```diff
@@ -41,36 +44,15 @@ void show_goto_functions(
   {
   case ui_message_handlert::XML_UI:
   {
-    //This only prints the list of functions
-    xmlt xml_functions("functions");
-    forall_goto_functions(it, goto_functions)
-    {
-      xmlt &xml_function=xml_functions.new_element("function");
-      xml_function.set_attribute("name", id2string(it->first));
-    }
-
-    std::cout << xml_functions << std::endl;
+    show_goto_functions_xmlt xml_show_functions(ns);
+    xml_show_functions.print_goto_functions(goto_functions, std::cout);
```

Comment:
I'd prefer operator(); otherwise `output`is more commonly used than `print` in the code base -- to be consistent, although I'd actually prefer print to output. Another argument is that the suffix `_goto_functions` is redundant because you are passing a goto_functionst as argument.